### ISSUE TITLE
Cppcheck may be invoked multiple times while on the fly inspection is in progress

### ISSUE THREAD

**Issue by firewave** [johnthagen/clion-cppcheck#33]
### Environment

* Operating System (e.g. Ubuntu 16.04 x64): Windows 10 Version 1909
* IDE Version (e.g. CLion 2016.3.2): Clion 2020.1 (201.6668.86)
* Cppcheck executable version (`cppcheck --version`): 1.90
* Cppcheck plugin version: 1.4.2
* Exact strings used in cppcheck plugin options:
  * **cppcheck path**:
  * **cppcheck options**: 

### Expected behaviour

Cppcheck is run only once per file in case it is not modified.

### Actual behaviour

Cppcheck is invoked every time the IDE regains focus.

### Steps to reproduce the behaviour

I have a file which takes extremely long to perform the on the fly inspections on (probably some issue causing this) and while the file is still being being running the inspections Cppcheck will be invoked every time the IDE regains focus (unless it is already running). This does not occur when the inspections are complete.

**Comment by firewave** [comment]
Appears the issue is slightly different than assumed - see https://youtrack.jetbrains.com/issue/CPP-22881

**Comment by firewave** [comment]
It seems the original issue also persists. I created another ticket with the CLion team - see https://youtrack.jetbrains.com/issue/CPP-22928
